app/src/main/java/org/schabi/newpipe/player/BackgroundPlayerActivity.java

```diff
@@ -56,14 +58,28 @@ public boolean onPlayerOptionSelected(MenuItem item) {
             }
 
             this.player.setRecovery();
-            getApplicationContext().sendBroadcast(getPlayerShutdownIntent());
-            getApplicationContext().startService(getSwitchIntent(PopupVideoPlayer.class));
+            NavigationHelper.playOnPopupPlayer(getApplicationContext(), player.playQueue, true);
             return true;
         }
+
+        if (item.getItemId() == R.id.action_switch_background) {
+            this.player.setRecovery();
+            NavigationHelper.playOnBackgroundPlayer(getApplicationContext(), player.playQueue, true);
+            return true;
+        }
+
         return false;
     }
 
     @Override
+    public void setupMenu(Menu menu) {
+        if(player == null) return;
+
+        menu.findItem(R.id.action_switch_popup).setVisible(!((VideoPlayerImpl)player).popupPlayerSelected());
+        menu.findItem(R.id.action_switch_background).setVisible(!((VideoPlayerImpl)player).audioPlayerSelected());
+    }
+
+    //@Override
```

Comment:
Because when audioPlayerSelected() == true  an audio player is already visible on the screen so no need to show `action_switch_background`. `Background` player == audio player. The same applies to a popup player: if the popup player is visible currently no need to show the menu item for it